File: hashmap/two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const map: Map<number, number> = new Map();

    for (let i = 0; i < nums.length; i++)
    {
        const complement = target - nums[i];
        if (map.has(complement))
        {
            return [i, map.get(complement)!];
        }

        if (!map.has(nums[i]))
        {
            map.set(nums[i],  i);
        }
    }

    return [];
};

File: hashmap/two-sum/two-sum.cs
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> dict = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++)
        {
            int complement = target - nums[i];
            if (dict.ContainsKey(complement))
            {
                return [i, dict[complement]];
            }

            if (!dict.ContainsKey(nums[i]))
            {
                dict[nums[i]] = i;
            }
        }

        return null;
    }
}